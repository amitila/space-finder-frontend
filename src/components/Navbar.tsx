import React from "react";
import { User } from "../model/Model";
import { Link } from "react-router-dom";

export class Navbar extends React.Component<{ user: User | undefined }> {
  render() {
    let loginLogOut: any;
    if (this.props.user) {
      loginLogOut = (
        <Link className="login" to="/logout">
          {this.props.user.userName}
        </Link>
      );
    } else {
      loginLogOut = (
        <Link data-testid="login-link" className="login" to="/login">
          Login
        </Link>
      );
    }

    return (
      <div className="navbar">
        <Link data-testid="home-link" className="link" to="/">
          Home
        </Link>
        <Link data-testid="profile-link" className="link" to="/profile">
          Profile
        </Link>
        <Link data-testid="spaces-link" className="link" to="/spaces">
          Spaces
        </Link>
        <span>{loginLogOut}</span>
      </div>
    );
  }
}
