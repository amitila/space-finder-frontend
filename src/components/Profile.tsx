import React from "react";
import { User, UserAttribute } from "../model/Model";
import { AuthService } from "../services/AuthService";
import { Link } from "react-router-dom";

interface ProfileProps {
  user: User | undefined;
  authService: AuthService;
}

interface ProfileState {
  userAttributes: UserAttribute[];
}

export class Profile extends React.Component<ProfileProps, ProfileState> {
  state: ProfileState = {
    userAttributes: [],
  };

  async componentDidMount() {
    if (this.props.user) {
      const userArts = await this.props.authService.getUserAttribute(
        this.props.user
      );
      this.setState({
        userAttributes: userArts,
      });
    }
  }

  private renderUserAttributes() {
    const rows = [];
    for (const userAttribute of this.state.userAttributes) {
      rows.push(
        <tr key={userAttribute.Name}>
          <td>{userAttribute.Name}</td>
          <td>{userAttribute.Value}</td>
        </tr>
      );
    }
    return (
      <table>
        <tbody>{rows}</tbody>
      </table>
    );
  }

  render() {
    let profileSpace;
    if (this.props.user) {
      profileSpace = (
        <div>
          <h3>Hello {this.props.user.userName}</h3>
          Here are your attributes:
          {this.renderUserAttributes()}
        </div>
      );
    } else {
      profileSpace = (
        <div>
          Please <Link to="login">Login</Link>
        </div>
      );
    }
    return <div>Welcome to Profile {profileSpace}</div>;
  }
}
