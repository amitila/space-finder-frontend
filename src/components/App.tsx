import React from "react";
import { User } from "../model/Model";
import { AuthService } from "../services/AuthService";
import { Login } from "./Login";
import { Router, Route, Switch } from "react-router-dom";
import { Navbar } from "./Navbar";
import { Home } from "./Home";
import { Profile } from "./Profile";
import { Spaces } from "./spaces/Spaces";
import { DataService } from "../services/DataService";
import history from "../utils/history";

interface AppState {
  user: User | undefined;
}

export class App extends React.Component<{}, AppState> {
  private authService: AuthService = new AuthService();
  private dataService: DataService = new DataService();

  constructor(props: any) {
    super(props);
    this.state = { user: undefined };
    this.setUSer = this.setUSer.bind(this);
  }

  private setUSer(user: User) {
    console.log("user ", user);
    this.setState({
      user: user,
    });
  }

  render() {
    return (
      <div className="wrapper">
        <Router history={history}>
          <div>
            <Navbar user={this.state.user} />
            <Switch>
              <Route exact path="/" component={Home} />
              <Route exact path="/login">
                <Login authService={this.authService} setUser={this.setUSer} />
              </Route>
              <Route exact path="/profile">
                <Profile
                  authService={this.authService}
                  user={this.state.user}
                />
              </Route>
              <Route exact path="/spaces">
                <Spaces dataService={this.dataService} />
              </Route>
            </Switch>
          </div>
        </Router>
      </div>
    );
  }
}
