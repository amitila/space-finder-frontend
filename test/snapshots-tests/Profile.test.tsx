import React from "react";
import { create } from "react-test-renderer";
import { Profile } from "../../src/components/Profile";
import { StaticRouter } from "react-router-dom";

describe("Profile component snapshot testing", () => {
  test("initial test", () => {
    const snap = create(
      <StaticRouter>
        <Profile user={undefined} authService={{} as any} />
      </StaticRouter>
    );
    expect(snap).toMatchSnapshot();
  });
});
