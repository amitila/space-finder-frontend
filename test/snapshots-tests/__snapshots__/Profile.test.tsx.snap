// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component snapshot testing initial test 1`] = `
<div>
  Welcome to Profile 
  <div>
    Please 
    <a
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </div>
</div>
`;
